### PR TITLE
Add script parameters to download books grabbed between given dates

### PR DESCRIPTION
**Pull request by mjenczmyk**
In this pull request I'm adding parameters allowing to download books between two given dates, I'm also removing possibility to download books by their name.

### CODE REVIEW

File: src/packtPublishingFreeEbook.py [hunk 1]
```diff
@@ -203,7 +206,15 @@ def __get_my_all_books_data(self):
                 for fm in self.download_formats:
                     if url.find(fm) != -1:
                         download_urls[fm] = url
-            self.book_data.append({'title': title, 'download_urls': download_urls})
+            order_date = parse(
```
Comment (mjenczmyk) [inline src/packtPublishingFreeEbook.py]: `datetime.strptime` won't work here - it depends on locales, that's why I'm using `dateutil.parse.parser`.

**Comment by mjenczmyk** [comment]
@RuthlessRuler, @igbt6 Could you review and verify that it works as expected?

**Comment by igbt6** [comment]
Hey, Finally had some time to take a look on these changes. I think there is no need for such features
What is the point ? Packt often repeats the same books, so if you have it already grabbed at your account, it will be not grabbed again and date parameters is not useful at all  here. You would have to maually search what was the order date of given book and then put into the script. The same feature which is  downloading some chosen titles. This option i would leave as it is, last time i used it :) (I wanted to download a given title in 4 formats, os i only copied the title and it's done, Someone might find it useful )

**Comment by RuthlessRuler** [comment]
@igbt6 The feature allows to download ebooks that were claimed between two days. If you use this page as reference: https://www.packtpub.com/account/my-order-history then you can easily find out the dates and download ebooks between them.

Sorry for such late reply btw.

**Comment by mjenczmyk** [comment]
OK, what should we do with that pull request? The requested feature is certainly more useful than `-dc` option, but I know I won't use any of them. 

@igbt6 Regardless of whether we add ability to download books grabbed between given dates should `-dc` option be removed?

**Comment by nevzat** [comment]
I'd certainly want that feature. I login and claim the books daily on my phone but download them from my pc periodically (let's say once in 2 weeks or a month). Currently I have to download-all and stop the script forcefully whenever I see the last downloaded book's name on the script output

**Comment by igbt6** [comment]
I will close this pull request - As i mentioned before, the script should be clean and simple, I find such features as really not useful

**Comment by nevzat** [comment]
@mjenczmyk is it possible to download this patch and apply to my own local version via git/github or do you have this patch applied on your version? I'm still a newby at git+github+python but I'm progressing.  If not possible I'll copy-paste the changes one by one, old fashion :)

**Comment by mjenczmyk** [comment]
Hi,

the feature you want is on my branch `rm`, but the script does not work currently. I've managed to fix downloading on my branch `setup` (but only downloading, grabbing still doesn't work). We have to wait until Packtpub sale will be over - only than I can try to fix it.

You can try to fix that by rebasing my `rm` branch with `setup` branch (that is by running `git rebase setup` on `rm` branch - you will have to add my repository as another remote) - can you try to do that? It may not be that easy, but still should be doable. I'll certainly do it after script will be fixed (I just want to avoid to doing that twice as `setup` branch will certainly change).

If you fail doing this let me know - I'll do that in my free time (but I really cannot promise it'll be before weekend).

**Comment by nevzat** [comment]
Many thanks for all the efforts and reply. I'll try it in my spare time soon. Meanwhile I also learned that if I add ".patch" to this url github allows downloading as patch. I still haven't tried implementing it but it's handy.

**Comment by mjenczmyk** [comment]
@nevzat Try `pip install git+https://github.com/mjenczmyk/Packt-Publishing-Free-Learning.git@bd --upgrade` and see if it works.

**Comment by nevzat** [comment]
thanks @mjenczmyk  . Currently I'm working on a GUI port for this script. I hope I can do it and share on my GitHub as my first project :)

**Pull request closed without merging**